### PR TITLE
Process Data: Generate extraction schema

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Add support for a model to generate the Process Data action schema.
- Adds link to a documentation page for Process action: https://dust-tt.notion.site/Process-Data-on-Dust-424a836b881e4664aa6077a344a55b0c

Screenshot from 2024-05-16 17-19-44
Screenshot from 2024-05-16 17-19-48
Screenshot from 2024-05-16 17-19-55

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  switch (req.method) {
+    case "POST":
+      const bodyValidation =
+        InternalPostBuilderProcessActionGenerateSchemaRequestBodySchema.decode(
+          req.body
+        );
+
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts:50]: we have [this](https://github.com/dust-tt/dust/blob/4295a6c4e5c3563e189fecd32bc13655f93a3ce0/types/src/shared/utils/iots_utils.ts#L47-L57) now

**Review by fontanierh — APPROVED** [review]
very cool

File: front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts (hunk 1 above)
Comment (spolu) [reply front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts:50]: Not much shorter but switching to it :)

Screenshot from 2024-05-16 17-36-17

Comment (fontanierh) [reply front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts]: True it's not that helpful :p

Comment (fontanierh) [reply front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts]: slightly more readable I guess

Comment (spolu) [reply front/pages/api/w/[wId]/assistant/builder/process/generate_schema.ts]: def

**Pull request merged**